server/store/encrypter.go

```diff
@@ -0,0 +1,88 @@
+package store
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"io"
+
+	"github.com/pkg/errors"
+)
+
+type Encrypter interface {
+	Encrypt(message []byte) ([]byte, error)
+	Decrypt(message []byte) ([]byte, error)
+}
+
+type AESEncrypter struct {
+	key []byte
+}
+
+var _ Encrypter = (*AESEncrypter)(nil)
+
+func (s *AESEncrypter) unpad(src []byte) ([]byte, error) {
+	length := len(src)
+	unpadding := int(src[length-1])
+
+	if unpadding > length {
+		return nil, errors.New("unpad error. This could happen when incorrect encryption key is used")
+	}
+
+	return src[:(length - unpadding)], nil
+}
+
+func (s *AESEncrypter) pad(src []byte) []byte {
+	padding := aes.BlockSize - len(src)%aes.BlockSize
+	padtext := bytes.Repeat([]byte{byte(padding)}, padding)
+	return append(src, padtext...)
+}
```

Comment:
Why do we need the padding? `crypto/cipher.Stream.XORKeyStream` doesn't have any length requirement to the `src`.

Reply:
I still don't understand where these additional characters come from. Does the encryption stream add them?

FWIW: The go example also doesn't use padding: https://pkg.go.dev/crypto/cipher#example-NewCFBDecrypter

Reply:
I changed it to base64 + CFBEncrypter/Decrypter, much more stable and easy to debug.